src/compiler/objective_c_generator.cc

```diff
@@ -176,65 +176,71 @@ void PrintMethodImplementations(Printer *printer,
 
 string GetHeader(const ServiceDescriptor *service, const string prefix) {
   string output;
-  grpc::protobuf::io::StringOutputStream output_stream(&output);
-  Printer printer(&output_stream, '$');
+  {
+    // Scope the output stream so it closes and finalizes output to the string.
```

Comment:
Oh, scratch most of that. The destruction order [is determined](http://stackoverflow.com/questions/14688285/c-local-variable-destruction-order).